### ISSUE TITLE
[QUALITY ASSURANCE]:  Testing Grading page (sprint 18)

### ISSUE THREAD

**Issue by Ravirkt** [fdnd-agency/footguard#218]
# TESTEN

Nogmaals controleren of alles echt werkt naar behoren:

- [x] Browser Test (Controleren of alles echt werkt op verschillende browsers)
- [x] Performance test
- [x] Toegankelijkheid - WCAG - a11y - screenreader - keyboard

**Comment by Ravirkt** [comment]
# Browser Test 🌐

Test uitgevoerd op de volgende browsers:  

- Safari 🧭  
- Chrome 🌏  
- Edge 🔵  
- Opera 🎤  
- Firefox 🔥  

---

## Testpunten ✅
- Dynamische data wordt correct ingeladen  
- Styling van de pagina is correct  
- Formulieren werken naar behoren  
- Overflow issues (voor Firefox)  

---

## Resultaten per browser 📊

### Safari 🧭
- Dynamische data correct geladen  
- Styling en form werken goed  
<img width="1899" height="1259" alt="Safari Test Screenshot" src="https://github.com/user-attachments/assets/0d64ddf2-357c-4725-a371-7187481a925b" />

### Chrome 🌏
- Dynamische data correct geladen  
- Styling en form werken goed  
<img width="1897" height="1172" alt="Chrome Test Screenshot" src="https://github.com/user-attachments/assets/14b70661-5ef5-46af-b7f8-a37d8cf5c263" />

### Edge 🔵
- Dynamische data correct geladen  
- Styling en form werken goed  
<img width="1898" height="1248" alt="Edge Test Screenshot" src="https://github.com/user-attachments/assets/8e7ffeb8-d04d-45c1-af15-c24dbaad8009" />

### Opera 🎤
- Dynamische data correct geladen  
- Styling en form werken goed  
<img width="1844" height="1225" alt="Opera Test Screenshot" src="https://github.com/user-attachments/assets/62b12b24-e785-421b-93d4-f3b1dfd33cdc" />

### Firefox 🔥
- Dynamische data correct geladen  
- Styling en form werken goed  
- **Opmerking:** veel overflow aanwezig, hiervoor is issue #219 aangemaakt  

https://github.com/user-attachments/assets/b82ee0bc-a6f4-4099-86ae-2c15b87e35ef

**Comment by Ravirkt** [comment]
# Performance test

## Desktop

<img width="1898" height="1161" alt="Image" src="https://github.com/user-attachments/assets/2a30bb9e-85fc-42c1-abfb-4f9b67ff1817" />

## Mobile

<img width="1486" height="1146" alt="Image" src="https://github.com/user-attachments/assets/7086ef18-ce51-4e9b-b5bc-6afedf29ba00" />

**Comment by Ravirkt** [comment]
# Toegankelijkheid ♿

Test uitgevoerd op de volgende onderdelen:  

- Screenreader 📢  
- Keyboard ⌨️  
- Lighthouse 🔍  
- A11y checklist 📋  

---

## Keyboard ⌨️

### Testpunten ✅
- De gebruiker kan door de pagina **tabben** zonder probleem  
- Duidelijke focus outline zichtbaar  
- form is te bedienen met toetsenbord 

### Resultaten 📊
- Alle interactieve elementen zijn toegankelijk met het toetsenbord  
- Tabvolgorde loopt logisch van boven naar beneden  
- De checklist kan volledig met toetsenbord worden ingevuld
- De dropdown kan met toetsenbord open en gesloten worden
- Je kunt tabben door de verschillende pfd functies en ook door de links in de pdf zelf.

---

## Lighthouse 🔍

### Testpunten ✅
- Desktop audit  
- Mobile audit  

### Resultaten 📊

#### Desktop

<img width="1161" height="1187" alt="Image" src="https://github.com/user-attachments/assets/2f4cd400-1ba7-445e-930d-a2ee8d4f14a4" />

#### Mobile

<img width="1159" height="1166" alt="Image" src="https://github.com/user-attachments/assets/41878d6a-99e0-4a47-822e-372fe69051f0" />

---

## Screenreader 📢

### Testpunten ✅
- Kan pagina correct voorlezen  
- Alle interactieve elementen zijn toegankelijk  

### Resultaten 📊
- De screenreader werkt naar behoren
- screenreader leest ook de functionaliteiten van de pdf reader op
- Demo / opname: [Bekijk hier](https://github.com/user-attachments/assets/43d96b9e-2c69-4daa-9cc8-c1f5d94ca575)

---

## A11y Checklist 📋

### Testpunten ✅
- Content, global code, keyboard, images, headings, lists, controls, tables, forms, media, appearance & color contrast, mobile & touch  

### Resultaten 📊
### Content
- [x] Use plain language and avoid figures of speech, idioms, and complicated metaphors.
  - gecontroleerd op grammatica, engelse taal en spelling
- [x] Make sure that `button`, `a`, and `label` element content is unique and descriptive.
  - naamgeving buttons , a, en labels zijn gecontroleerd op functie doeleinde 
- [x] Use left-aligned text for LTR languages, and right-aligned text for RTL languages.
  - uitlijning tekst is correct begint van links naar rechts

### Global code
- [x] Validate your HTML.
  - https://github.com/fdnd-agency/footguard/issues/69#issuecomment-3464465422
- [x] Use a `lang` attribute on the `<html>` element.
  - <html lang="en"> is aanwezig
- [x] Ensure that zoom is not disabled.
  - de gebruiker kan in en- en uitzoomen in de pagina
- [x] Use landmark elements (`<nav>`, `<main>`, `<footer>`).
  - footer is niet aanwezig omdat deze niet relevant is
- [x] Ensure linear content flow.
  - de interactieve elementen zijn toegankelijk op logische flow van de pagina
- [x] Avoid `autofocus` and make session timeouts extendable.
  - geen gebruik gemaakt van autofocus
- [x] Use labels instead of relying on tooltips.
  - buttons hebben een label en worden hidden gemaakt met: https://www.a11yproject.com/posts/how-to-hide-content/
   Het is nu dus nog wel toegankelijk met screenreader

### Keyboard
- [x] All interactive elements (links, buttons, controls) are operable via keyboard.
  - de interactieve elementen zijn toegankelijk met de toetsenbord. je kunt van boven naar beneden tabben per element. Je kunt een value selecteren in de selectbuttons met spacebar en doorgaan naar een andere pagina met de enter button
- [x] There is a visible focus-style for interactive elements.
  - De interactieve elementen hebben alle een duidelijke outline dat zichtbaar is waarop gefocust is.

### Images
- [x] All meaningful images have appropriate alt text.
  - Er zijn geen images op de pagina alleen icons
- [x] Avoid using color as the only way to convey information.
  - er is voornamelijk met een icon, text en kleur aangegeven

### Headings
- [x] Use headings in logical order (`<h1>` → `<h2>` → `<h3>`).
  - headings zijn op logische volgorde gebruikt
- [x] Do not skip heading levels.
  - headings levels zijn niet overgeslagen alles begint van h1 naar aflopend

### Lists
- [x] Use correct list markup: `<ul>`, `<ol>`, or `<dl>` where appropriate.
  - n.v.t
- [x] Ensure the list structure makes sense for assistive technologies.
  - n.v.t

### Controls
- [x] Buttons, links and form controls have clear, descriptive labels.
  - buttons en labels hebben duidelijke benaming en label naam
- [x] Interactive elements that are not standard controls must provide proper ARIA roles or labels.
  - n.v.t

### Tables
- [x] Use `<th>` for headers and ensure clear association with `<td>` cells.
  - n.v.t
- [x] Do not use tables purely for layout.
  - n.v.t

### Forms
- [x] Ensure each form control has a `<label>` (explicit or via `aria-label`).
  - de form heeft voor de inputs alle een label
- [ ] Error messages should be clearly visible and explain how to fix the error.
  - de post van de form is nog niet werkend

### Media (Video / Audio)
- [x] Provide captions or transcripts for videos and audios.
  - n.v.t
- [x] Media controls must be accessible (keyboard, screen-reader).
  - n.v.t

### Appearance & Color contrast
- [x] Color is not the only way to convey information.
  - zowel kleur als text en icons maken informatie duidelijk
- [x] Maintain sufficient contrast between text and background (WCAG AA: 4.5:1 for normal text).
  - alle kleuren zijn getest op contrast zie: https://github.com/fdnd-agency/footguard/issues/69#issuecomment-3464598630
- [x] Ensure icons and badges also meet contrast criteria.
  - de icons zijn getest op contrast zie: https://github.com/fdnd-agency/footguard/issues/69#issuecomment-3464598630

### Mobile & Touch
- [x] Touch-targets (buttons/links) are large enough and spaced adequately.
  - De interactieve elementen zijn groot genoeg en gemakkelijk klikbaar op touchscreen devices
- [x] Users can zoom and rotate screen without losing access to content.
  - Getest en werkt
